src/hotspot/share/oops/instanceKlass.hpp

```diff
@@ -229,7 +229,9 @@ class InstanceKlass: public Klass {
   // _idnum_allocated_count.
   volatile ClassState _init_state;          // state of class
 
-  u1              _reference_type;                // reference type
+  u1                 _reference_type;                // reference type
+
+  AccessFlags        _access_flags;    // Access flags. The class/interface distinction is stored here.
 
   // State is set either at parse time or while executing, atomically to not disturb other state
   InstanceKlassFlags _misc_flags;
```

Comment:
It looks like the size of these 3 fields, _reference_type, _access_flags, and _misc_flags, will take 8 bytes because of alignment/padding, but if _reference_type was moved to the end of InstanceKlassFlags and the order reversed, it could be reduced to 6 bytes, leaving 2 for future expansion:
```
InstanceKlassFlags _misc_flags; // contains _reference_type
AccessFlags        _access_flags;
char _reserved[2];
```

Reply:
You're right, I missed that.